### PR TITLE
Add body filtering to Tesla Middleware

### PR DESCRIPTION
**Pull request by nvieirafelipe**

### CODE REVIEW

**Review by zekus — APPROVED** [review]

**Review by dsnipe — APPROVED** [review]

File: lib/tesla/middleware/meta_logger.ex [hunk 1]
```diff
@@ -24,6 +25,12 @@ if Code.ensure_loaded?(Tesla) do
     the values will be replaced with `[FILTERED]`, defaults to: `[]`.
     * `:filter_query_params` - The query params that should not be logged,
     the values will be replaced with `[FILTERED]`, defaults to: `[]`
+    * `:filter_body` - The request and response body patterns that should not be logged,
```
Comment (prodis) [inline lib/tesla/middleware/meta_logger.ex:28]: Please add this information also to https://github.com/FindHotel/meta_logger/tree/29a568ad9902d8d693ec73cc7f0edb4e3476a669#options.

File: lib/tesla/middleware/meta_logger.ex [hunk 2]
```diff
@@ -116,6 +128,18 @@ if Code.ensure_loaded?(Tesla) do
     defp filter_query_params({key, _value} = param, filter_query_params),
       do: if(key in filter_query_params, do: {key, @filtered}, else: param)
 
+    @spec build_body(Env.body(), Env.opts()) :: Env.body()
+    defp build_body(body, options) when is_binary(body) do
+      filter_body = Keyword.get(options, :filter_body)
+      Enum.reduce(filter_body, body, &filter_body(&2, &1))
```
Comment (prodis) [inline lib/tesla/middleware/meta_logger.ex:134]: Suggestion:

```elixir
options
|> Keyword.get(:filter_body)
|> Enum.reduce(body, &filter_body(&2, &1))
```

**Review by prodis — APPROVED** [review]

**Review by sergioaugrod — APPROVED** [review]

File: lib/meta_logger/slicer.ex [hunk 3]
```diff
@@ -0,0 +1,46 @@
+defmodule MetaLogger.Slicer do
+  @moduledoc """
+  Responsible for slicing log entries according to the given the max length option.
+  """
+
+  @typedoc "Max length in bytes or `:infinity` if the entry should not be sliced."
+  @type max_entry_length :: non_neg_integer() | :infinity
+
+  @doc """
+  Returns sliced log entries sccording to the given max length option.
```
Comment (sergioaugrod) [inline lib/meta_logger/slicer.ex:10]:
```suggestion
  Returns sliced log entries according to the given max length option.
```

File: lib/meta_logger/slicer.ex [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+      iex> #{inspect(__MODULE__)}.slice("1234567890", 5)
+      ["12345", "67890"]
+
+  """
+  @spec slice(String.t(), max_entry_length()) :: [String.t()]
+  def slice(entry, max_entry_length)
+      when max_entry_length == :infinity
+      when byte_size(entry) <= max_entry_length,
+      do: [entry]
```
Comment (dsnipe) [inline lib/meta_logger/slicer.ex:32]: What do you think if to make `:infinity` as a default value as well?

File: lib/meta_logger/slicer.ex (hunk 3 above)
Comment (sergioaugrod) [inline lib/meta_logger/slicer.ex:10]:
```suggestion
  Returns sliced log entries according to the given max entry length.
```

File: test/meta_logger/slicer_test.exs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      {:ok, entry: entry}
+    end
+
+    test "when `:infinity` is given as max entry length, returns a list with one entry", %{
+      entry: entry
+    } do
+      assert Subject.slice(entry, :infinity) == [entry]
+    end
+
+    test "when max entry length is smaller than the size of given entry, returns a list one entry",
```
Comment (sergioaugrod) [inline test/meta_logger/slicer_test.exs:21]:
```suggestion
    test "when max entry length is smaller than the size of given entry, returns a list with one entry",
```

File: lib/meta_logger/slicer.ex (hunk 4 above)
Comment (nvieirafelipe) [reply lib/meta_logger/slicer.ex:32]: It is, check the previous lines. We set the default options already

File: .github/workflows/elixir.yml [hunk 6]
```diff
@@ -25,8 +25,8 @@ jobs:
       - name: Setup Elixir
         uses: erlef/setup-beam@v1
         with:
-          elixir-version: '1.10.4'
-          otp-version: '23.0.4'
+          elixir-version: "1.10.4"
```
Comment (prodis) [inline .github/workflows/elixir.yml:28]: Out of the scope of this PR: the CI should run in all the (major) supported Elixir versions.

File: lib/meta_logger/slicer.ex [hunk 7]
```diff
@@ -0,0 +1,46 @@
+defmodule MetaLogger.Slicer do
+  @moduledoc """
+  Responsible for slicing log entries according to the given the max length option.
```
Comment (prodis) [inline lib/meta_logger/slicer.ex:3]:
```suggestion
  Responsible for slicing log entries according to the given max length option.
```

**Pull request merged**